core/corehttp/gateway_test.go

```diff
@@ -65,10 +66,19 @@ func newNodeWithMockNamesys(t *testing.T, ns mockNamesys) *core.IpfsNode {
 func TestGatewayGet(t *testing.T) {
 	ns := mockNamesys{}
 	n := newNodeWithMockNamesys(t, ns)
-	k, err := coreunix.Add(n, strings.NewReader("fnord"))
+	dagNode, err := importer.BuildDagFromReader(
+		strings.NewReader("fnord"),
+		n.DAG,
+		n.Pinning.GetManual(),
+		chunk.DefaultSplitter)
```

Comment:
:-1: I'm not ok with this change. This forces all those locations to understand what (a) a dagservice is, (b) what pinning is, (c) what chunking is. This is moving up a ton of complexity in places it does not belong. The ability to sink down and do this is good to have and should continue to be there, but _removing a useful_ abstraction is not good. Particularly when users of ipfs as a library will be looking at our code and following suit.